crates/parser/src/grammar/items.rs

```diff
@@ -414,10 +430,14 @@ fn fn_(p: &mut Parser<'_>, m: Marker) {
     m.complete(p, FN);
 }
 
-fn macro_call(p: &mut Parser<'_>) -> BlockLike {
+fn macro_call(p: &mut Parser<'_>, m: Marker) {
     assert!(paths::is_use_path_start(p));
     paths::use_path(p);
-    macro_call_after_excl(p)
+    match macro_call_after_excl(p) {
+        BlockLike::Block => p.eat(SEMICOLON),
+        BlockLike::NotBlock => p.expect(SEMICOLON),
+    };
+    m.complete(p, MACRO_CALL);
```

Comment:
Let's undo this change here again, this is semantically different from the previous behavior. `foo!{};` is not a valid item parse

Reply:
Done

Reply:
But wait. Was your coment about not doing the `eat` or about not having it as `macro_call(p: &mut Parser<'_>, m: Marker)` or about both? Because I very much prefer the to add the marker to the signature to be consistent with the other functions.